src/core/stompest/sync/client.py

```diff
@@ -136,7 +136,11 @@ def _connect(self, headers, versions, host, heartBeats, timeout):
         frame = self.session.connect(self._config.login, self._config.passcode, headers, versions, host, heartBeats)
         self.sendFrame(frame)
         if not self.canRead(timeout):
-            self.session.disconnect()
```

Comment:
I believe this line should simply be replaced by `self.session.close()`.

Please add a test case which breaks in the old version and passes in the new version.